### PR TITLE
[ASSIST-821]: Email Templating Engine

### PR DESCRIPTION
**Pull request by mxm1070**
### Ticket(s) or GitHub Issue

https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/821/

### Technical Description

Email templates base.

### Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] New feature (non-breaking change which adds functionality)
- [x] Content or styling update (Changes which don't affect functionality)

### Any deployment steps required?

A deployment step could be a command that needs to be executed or an ENV key that needs to be added, for example.

- [x] No

_______________________________________________

## Before contributing and submitting this PR, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/canyongbs/assistbycanyongbs/blob/main/README.md#contributing).
* [x] Title the PR with the ticket/issue number and a short description of the changes made. Or if no ticket/issue exists, title the PR with a short description of the changes made
* [x] Linked a relevant ticket or issue or describe the issue/feature which this PR resolves/implements.
* [x] Resolved all conflicts, if any.
* [x] Rebased your branch PR on top of the latest upstream `develop` branch.

### CODE REVIEW

File: .phpactor.json
Comment (Orrison) [inline .phpactor.json:1]: What is the purpose of this file?

File: app/Filament/Resources/EmailTemplateResource/Pages/EditEmailTemplate.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class EditEmailTemplate extends EditRecord
+{
+    protected static string $resource = EmailTemplateResource::class;
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                TextInput::make('name')
```
Comment (Orrison) [inline app/Filament/Resources/EmailTemplateResource/Pages/EditEmailTemplate.php:22]: What do you think about giving the option to edit the `model`? Is there a reason you didn't have it here?

File: app/Filament/Resources/EmailTemplateResource/Pages/EditEmailTemplate.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    ->string()
+                    ->required()
+                    ->autocomplete(false),
+                Select::make('primary_color')
+                    ->native(false)
+                    ->allowHtml()
```
Comment (Orrison) [inline app/Filament/Resources/EmailTemplateResource/Pages/EditEmailTemplate.php:28]: Why allowHtml?

File: app/Filament/Resources/EmailTemplateResource/Pages/ListEmailTemplates.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    protected static string $resource = EmailTemplateResource::class;
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                TextColumn::make('name'),
+                TextColumn::make('model.name')
+                    ->label('Related To')
+                    ->getStateUsing(fn (EmailTemplate $record): ?string => $record->model->name)
```
Comment (Orrison) [inline app/Filament/Resources/EmailTemplateResource/Pages/ListEmailTemplates.php:28]: You are probably going to want to do the same thing you are doing for `url` there is no guarantee that all future models will have `name`

File: app/Models/EmailTemplate.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        'model_type',
+    ];
+
+    public function registerMediaCollections(): void
+    {
+        $this->addMediaCollection('logo')
+            ->singleFile();
+    }
+
+    public function model(): MorphTo
```
Comment (Orrison) [inline app/Models/EmailTemplate.php:26]: I feel like the term `model` is a little to generate for what this is.

Do you have any ideas for terms that would better encapsulate this? consumer, appliedTo, etc.

File: routes/web.php [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
 Route::get('/imports/{import}/failed-rows/download', DownloadImportFailureCsv::class)
     ->name('imports.failed-rows.download')
     ->middleware(['auth']);
 
-Route::get('form-test', function () {
+Route::get('/form-test', function () {
     return view('form-test');
 });
+
+Route::get('/demo-notification', function () {
```
Comment (Orrison) [inline routes/web.php:16]: Let's add a TODO to delete this once the feature is finished.

**Review by Orrison — CHANGES_REQUESTED** [review]
Looking good. I added some questions and comments.

Please take the list you posted in the ticket and expand upon each Notification, listing out what you think are the possible pathways to template resolution for each.
We will then review it, and then pass it off to Product to make the decision for each.

File: .phpactor.json
Comment (mxm1070) [reply .phpactor.json:1]: It's from Tinkerwell. I thought I added it to the .gitignore but it snuck it anyway.

File: app/Filament/Resources/EmailTemplateResource/Pages/EditEmailTemplate.php (hunk 2 above)
Comment (mxm1070) [reply app/Filament/Resources/EmailTemplateResource/Pages/EditEmailTemplate.php:28]: I was playing with a color indicator and I forgot to remove it after I scrapped the idea.

File: app/Models/EmailTemplate.php (hunk 4 above)
Comment (mxm1070) [reply app/Models/EmailTemplate.php:26]: What about?
```php
public function relatedTo(): MorphTo
{
    return $this->morphTo('model');
}
```
Do you want me to to change the database columns?

Comment (Orrison) [reply app/Models/EmailTemplate.php]: That works, yeah if you could, please.

**Review by Orrison — APPROVED** [review]
Merging this in for now. Ticket completion pending on getting answers from Product though.

**Pull request merged**